nway/image_processing_utils.py

```diff
@@ -0,0 +1,145 @@
+import cv2
+import numpy as np
+import logging
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def register_image_pair(img_fixed: np.ndarray, img_moving: np.ndarray,
+                        maxCount: int, epsilon: float,
+                        motion_type: str, gaussFiltSize: int,
+                        preregister: bool = True) -> np.ndarray:
+    """find the transform that registers two images
+
+    Parameters
+    ----------
+    img_fixed : numpy.ndarray
+        CV_8U (uint8) or CV_32F (float32) fixed image
+    img_moving : numpy.ndarray
+        CV_8U (uint8) or CV_32F (float32) moving image
+    maxCount : int
+        passed as maxCount to opencv termination criteria
+    epsilon : float
+        passed as epsilon to opencv termination criteria
+    motion_type : str
+        one of the 4 possible motion types for opencv findTransformECC
+        see the dictionary cvmotion below for the 4 possible values
+    gaussFiltSize : int
+        passed to opencv findTransformECC(). An optional value
+        indicating size of gaussian blur filter
+    preregister: bool
+        whether to give a hint to cv2.findTransformECC from cv2.phaseCorrelate
+
+    Returns
+    -------
+    tform : :class:`numpy.ndarry`
+        3 x 3 transformation matrix
+
+    """
+
+    cvmotion = {
+            "MOTION_TRANSLATION": cv2.MOTION_TRANSLATION,
+            "MOTION_EUCLIDEAN": cv2.MOTION_EUCLIDEAN,
+            "MOTION_AFFINE": cv2.MOTION_AFFINE,
+            "MOTION_HOMOGRAPHY": cv2.MOTION_HOMOGRAPHY}
+
+    # Define termination criteria
+    criteria = (
+            cv2.TERM_CRITERIA_EPS | cv2.TERM_CRITERIA_COUNT,
+            maxCount,
+            epsilon)
+
+    dx = dy = 0.0
+    if preregister:
+        (dx, dy), _ = cv2.phaseCorrelate(img_fixed.astype('float32'),
+                                         img_moving.astype('float32'))
+
+    tform = np.array([[1.0, 0.0, dx],
+                      [0.0, 1.0, dy]]).astype('float32')
+    if motion_type == 'MOTION_HOMOGRAPHY':
+        hrow = np.array([0.0, 0.0, 1.0]).astype('float32')
+        tform = np.vstack((tform, hrow))
+
+    try:
+        # Run the ECC algorithm. The results are stored in warp_matrix.
+        ccval, tform = cv2.findTransformECC(
```

Comment:
This might benefit from specifying explicit parameters maybe?

```
ccval, tform = cv2.findTransformECC(
    templateImage=img_fixed,
    inputImage=img_moving,
    warpMatrix=tform,
    motionType=cvmotion[motion_type],
    criteria=criteria,
    inputMask=None,
    gaussFiltSize=gaussFiltSize)

Reply:
added explicit parameters for all opencv calls in image processing